src/main/java/core/basesyntax/Calculator.java

```diff
@@ -0,0 +1,12 @@
+package core.basesyntax;
+
+public class Calculator {
+
+    public void calculateStatistics(Report report, ReportItem reportItem) {
+        if (Name.BUY.equals(reportItem.getName())) {
+            report.addBuy(reportItem.getValue());
+        } else {
+            report.addSupply(reportItem.getValue());
+        }
```

Comment:
The method `calculateStatistics` should be better named to reflect its purpose more clearly. Method names should be verbs or verb phrases. For example, `updateReportWithItem` might be a more informative name.